src/Features/Core/Portable/Extensions/ExtensionFolder.cs

```diff
@@ -28,7 +28,7 @@ private sealed class ExtensionFolder
             /// <summary>
             /// Lazily computed assembly loader for this particular folder.
             /// </summary>
-            private readonly AsyncLazy<(IAnalyzerAssemblyLoaderInternal? assemblyLoader, Exception? extensionException)> _lazyAssemblyLoader;
+            private readonly AsyncLazy<(IExtensionAssemblyLoader? assemblyLoader, Exception? extensionException)> _lazyAssemblyLoader;
```

Comment:
had to abstract out this type so we could test the varying things it could do with mocks.